### ISSUE TITLE
[alert_sender/receiver] alert ping corner case found by fpv

### ISSUE THREAD

**Issue by cindychip** [lowRISC/opentitan#3969]
Hi Michael,

After `alert_req_i` and `alert_ack_o` added to RTL, I updated the FPV testbench and found the corner case about alert ping as below:

This happened when I ran `alert_rx_tx_fpv` and the fired assertion is: 
```
  `ASSERT(AlertPing_A, !error_present ##1 $rose(ping_req_i) |->
      ##[1:9] ping_ok_o, clk_i, !rst_ni || error_present)
```
The issue here is when ping timer triggered the same ping response one clock cycle apart from each other, then even though the alert_handshake was finished for the first ping response, the `ping_ok` signal won't be asserted. Because [here](https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_alert_receiver.sv#L123) right when `ping_ok_o` will be set to 1, the following ping_request set the `ping_pending_q` to 0. 

I think this probably won't happen in the current setting of LFSR seed, but just want to report this so that you are aware of this corner case, and maybe we can mention that in the spec if you want to waive it (please let me know if you have already mentioned it in the script.. Could be I totally missed it!)

![image](https://user-images.githubusercontent.com/11466553/97606307-ddcaa480-19cc-11eb-8065-de772b136b14.png)

Thanks for your time and help.
Cindy

**Comment by cindychip** [comment]
This is the draft PR if you want to run it locally https://github.com/lowRISC/opentitan/pull/3970. You can do `./fpv prim_alert_rxtx_fpv -gui`

**Comment by msfschaffner** [comment]
Hm... I see. Maybe we need to align this assertion then because I think this failure is unrelated to the `alert_req_i` and `alert_ack_o` change. I.e. we probably have to add an assumption here that tells the tool that `ping_req_i` is going to stay asserted until `ping_ok_o` is asserted...

**Comment by cindychip** [comment]
Oh You are right, I missed the FSM in alert_handler_ping_timer to ensure handshake was done before sending another ping. I will add a FPV assumption for this. This should be a FPV issue! Thanks Michael for confirming. Interesting it did not show up before...

**Comment by cindychip** [comment]
Another corner cases that we decided to waive on FPV:
There is an extra ping_response one cycle after when ping_sig_int_err is recovered. This will be detected as a spurious ping. If a ping_req_i collides with this handshake, it will cause a ping_timeoue. Due to the increase complexity of the RTL, we decided to waive it.
 
![image](https://user-images.githubusercontent.com/11466553/97912375-93fdf900-1d01-11eb-8ccd-4599fd8e01d3.png)
